pkg/webhook/server/generic/pod/gracedelete/webhook.go

```diff
@@ -49,42 +49,44 @@ func (gd *GraceDelete) Name() string {
 func (gd *GraceDelete) Validating(ctx context.Context, c client.Client, oldPod, newPod *corev1.Pod, operation admissionv1.Operation) error {
 	// GraceDeleteWebhook FeatureGate defaults to false
 	// Add '--feature-gates=GraceDeleteWebhook=true' to container args, to enable gracedelete webhook
-	if !feature.DefaultFeatureGate.Enabled(features.GraceDeleteWebhook) || operation != admissionv1.Delete {
+	if !feature.DefaultFeatureGate.Enabled(features.GraceDeleteWebhook) || operation != admissionv1.Delete || !utils.ControlledByKusionStack(oldPod) {
```

Comment:
how about changing the `oldPod` in `utils.ControlledByKusionStack(oldPod)` to `newPod`?

Reply:
There is no `newPod` for DELETE request, since the oldObject field is `nil`.